var/spack/repos/builtin/packages/cxx/package.py

```diff
@@ -34,8 +35,13 @@ def test(self):
             cxx_opts = [compiler.cxx11_flag] if "c++11" in test else []
 
             cxx_opts += ["-o", exe_name, filepath]
-            compiled = self.run_test(cxx_exe, options=cxx_opts, installed=True)
+            cxx_exe = which(join_path(self.prefix.bin, cxx_exe))
+            compiled = cxx_exe(*cxx_opts)
 
             if compiled:
                 expected = ["Hello world", "YES!"]
-                self.run_test(exe_name, expected=expected)
+                exe_run = which(join_path(self.prefix.bin, exe_name))
+                out = exe_run(output=str.split, error=str.split)
+                assert expected in out
+            else:
+                assert False, "Did not compile"
```

Comment:
Recall that one of the goals is to tighten the code.  So the pattern to follow is more like https://spack.readthedocs.io/en/latest/packaging_guide.html#adding-custom-files except for each of the test executables.